FF/Lib.lean

```diff
@@ -0,0 +1,147 @@
+import FF.BitVec
+
+namespace FF
+
+class Field (K : Type _) extends Neg K, Add K, Sub K, Mul K where
+  /-- The zero element of the field, the additive identity. -/
+  zero : K
+  /-- The one element of the field, the multiplicative identity. -/
+  one : K
+  /-- Squares this element. -/
+  square : K → K
+  /-- Doubles this element. -/
+  double : K → K
+  /-- Computes the multiplicative inverse of this element.
+    Following mathlib convention, we do not fail on 0.
+    
+    TODO: Maybe should be `K → Option K` -/
+  inv : K → K
```

Comment:
Every element in a field has inverse, so it must be `K -> K`

Reply:
0 doesn't have an inverse